### ISSUE TITLE
Release failures

### ISSUE THREAD

**Issue by pavolloffay** [jaegertracing/jaeger-client-java#390]
Travis build: https://travis-ci.org/jaegertracing/jaeger-client-java/jobs/365086226#L1333 

Errors when repsitory is closed manually:
```
failureMessage | Missing SHA1: '/io/jaegertracing/jaeger-thrift/0.27.0-RC1/jaeger-thrift-0.27.0-RC1-sources.jar.sha1'
-- | --
failureMessage | Missing MD5: '/io/jaegertracing/jaeger-thrift/0.27.0-RC1/jaeger-thrift-0.27.0-RC1-sources.jar.md5'
failureMessage | Missing SHA1: '/io/jaegertracing/jaeger-thrift/0.27.0-RC1/jaeger-thrift-0.27.0-RC1.pom.sha1'
failureMessage | Missing MD5: '/io/jaegertracing/jaeger-thrift/0.27.0-RC1/jaeger-thrift-0.27.0-RC1.pom.md5'
failureMessage | Missing SHA1: '/io/jaegertracing/jaeger-core/0.27.0-RC1/jaeger-core-0.27.0-RC1.jar.sha1'
failureMessage | Missing MD5: '/io/jaegertracing/jaeger-core/0.27.0-RC1/jaeger-core-0.27.0-RC1.jar.md5'
failureMessage | Missing SHA1: '/io/jaegertracing/jaeger-crossdock/0.27.0-RC1/jaeger-crossdock-0.27.0-RC1.jar.sha1'
failureMessage | Missing MD5: '/io/jaegertracing/jaeger-crossdock/0.27.0-RC1/jaeger-crossdock-0.27.0-RC1.jar.md5'
```

**Comment by pavolloffay** [comment]
The publish failed even after #391  https://travis-ci.org/jaegertracing/jaeger-client-java/jobs/365101107#L1331

**Comment by pavolloffay** [comment]
The problem might be that publish creates 7 staging repositories. @jpkrohling  @yurishkuro have you seen this before?

**Comment by jpkrohling** [comment]
I've seen it before when releasing via Travis. If you checkout the tag locally and run `./gradlew upload`, it usually works fine. Just make sure you clean up any garbage created by Travis (ie: close/drop those staging repositories).

**Comment by pavolloffay** [comment]
I have tried the build 3 times, always it fails. I wondering if it even works sometimes automatically.

**Comment by jpkrohling** [comment]
I'm sure it used to work. Perhaps Travis is just having a bad day? :)

**Comment by jpkrohling** [comment]
In any case: locally running `./gradlew upload` should work.

**Comment by pavolloffay** [comment]
Please report if it does not work, when you are releasing. So we can fix it for future runs.

> I'm sure it used to work. Perhaps Travis is just having a bad day? :) 

Not sure, I am having a bad day that it does not work.

**Comment by pavolloffay** [comment]
Locally it also failed to close and release.

```
> Task :closeRepository 
Requested operation wasn't successful in first try. Retrying maximum 50 times with 3 seconds delay between attempts.
Requested operation was executed successfully in attempt 9 (maximum allowed 51)

> Task :releaseRepository FAILED
POST request failed. 500: Server Error, body: [errors:[[id:*, msg:Unhandled: Repository: iojaegertracing-1018 has invalid state: open]]]
```

**Comment by jpkrohling** [comment]
> Requested operation was executed successfully in attempt 9 (maximum allowed 51)

Can you check if the repository is there, in `closed` state? If so, you might just need to `release` it. I have a feeling that this Nexus plugin for Gradle might be outdated (or just broken)...

**Comment by pavolloffay** [comment]
I have closed it and released manually via UI

**Comment by pavolloffay** [comment]
I have opened an issue there https://github.com/Codearte/gradle-nexus-staging-plugin/issues/69

**Comment by pavolloffay** [comment]
The issue from travis could be this one https://github.com/researchgate/gradle-release/issues/244#issuecomment-372794864. 

> From what I see it's not directly related to the release plugin. The problem is the gradle internal on uploading the artifacts as it is done in multiple requests and if they use different outbound ip adresses the nexus is splitting these artifacts up.

**Comment by jpkrohling** [comment]
Related to #58

**Comment by pavolloffay** [comment]
Release of 0.28.0 via pushing tag on travis failed. Gradle publish created multiple staging repos see 

![screenshot of nexus repository manager](https://user-images.githubusercontent.com/5618424/40478937-aaf05472-5f4a-11e8-9994-2583f9f04a4a.jpg)

**Comment by pavolloffay** [comment]
0.29.0 also failed

**Comment by pavolloffay** [comment]
I think the issue could be solved by defining `stagingProfileId` see https://github.com/Codearte/gradle-nexus-staging-plugin#quick-start and https://issues.sonatype.org/browse/OSSRH-19485?focusedCommentId=340094&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-340094.

I am not sure if the ID has to change between releases or it can be hardcoded.

**Comment by pavolloffay** [comment]
My last attemp to use profile id failed. I have reported the issue in the plugin https://github.com/Codearte/gradle-nexus-staging-plugin/issues/76.

**Comment by pavolloffay** [comment]
Another solution is to do a release without any plugin like it was done in this project https://github.com/h2oai/sparkling-water/tree/master/gradle/publish and https://github.com/h2oai/sparkling-water/blob/73d2e2bd11ce35c6379f6961188fb2af1a7bc04b/build.gradle#L228.  It publishes to local folder and then publish to pre created repo.

**Comment by pavolloffay** [comment]
There is a new gradle plugin https://github.com/marcphilipp/nexus-publish-plugin which explicitly creates a staging repository. We should give it a try. It will require to bump a gradle version and use maven-publish plugin instead of legacy `maven` plugin we are using right now.

I will give it a try.

**Comment by pavolloffay** [comment]
Migration to new publish plugin didn't fix releasing from travis: https://travis-ci.org/jaegertracing/jaeger-client-java/jobs/526098222. I had to release locally.

**Comment by pavolloffay** [comment]
Fixed in #621
